lucene/core/src/java/org/apache/lucene/index/IndexWriter.java

```diff
@@ -1072,7 +1073,14 @@ public IndexWriter(Directory d, IndexWriterConfig conf) throws IOException {
 
       config.getFlushPolicy().init(config);
       bufferedUpdatesStream = new BufferedUpdatesStream(infoStream);
-      docWriter = new DocumentsWriter(flushNotifications, segmentInfos.getIndexCreatedVersionMajor(), pendingNumDocs,
+      final IntConsumer reserveDocs = numDocs -> {
+        if (numDocs > 0) {
+          reserveDocs(numDocs);
```

Comment:
I'm confused why we are calling separate methods when `numDocs` is negative or not?